lua/autorun/server/wall_kick_parkour.lua

```diff
@@ -1,139 +1,141 @@
-		
-		// Disregard this whole thing if the player is holding both left and right
-		if ( ply:KeyDown( IN_MOVELEFT ) and ply:KeyDown( IN_MOVERIGHT ) ) then return end
-        if ply:InVehicle() then return end
-        if ply:GetMoveType() == MOVETYPE_NOCLIP then return end
-        
-        // This determines which direction the player is trying to jump and sets dir accordingly
-        if ply:KeyDown( IN_MOVELEFT ) then
-            dir = 1
-        elseif ply:KeyDown( IN_MOVERIGHT ) then
-            dir = -1
-        end
+    if key ~= IN_JUMP then return end
+    if not wallkickEnabled then return end
 
-        // Due to overwhelming popular demand
-        if GetConVar( "wallkick_inverted" ):GetBool() then
-            dir = -dir
-        end
-        
-        // If we're trying to go left or right
-        if dir != 0 then
-
-            // Perform several traces to make it less likely that a hole in geometry would prevent a kick off
-            local traceCount = 5
-            local accumulatedNormal = Vector( 0, 0, 0 )
-            local hit = false
-            local hitCount = 0
-            for i = 0, ( traceCount - 1 ) do
-
-                // Our traces are from the bottom of the player to partway up their full height
-                local traceStartPos = ply:GetPos() + Vector( 0, 0, ply:OBBMaxs().z * ( 1/4 * i ) )
-
-                local tr = util.TraceLine( {
-                    start = traceStartPos,
-                    endpos = traceStartPos + ( ply:GetRight() * traceDistance * dir ),
-                    filter = ply,
-                } )
-                
-                
-                if tr.Hit then
-                    hit = true
-                    accumulatedNormal = accumulatedNormal + tr.HitNormal
-                    hitCount = hitCount + 1
-                end
+    local kickHeight = wallkickForceVertical or 250
+    local kickDistance = wallkickForceHorizontal or 150
+
+    if ply:IsOnGround() or ply.WallKickJumpedAlready then return end
+
+    -- By default our direction of jump is 0 which will make it easy to check later if it has changed
+    local dir = 0
+
+    -- Disregard this whole thing if the player is holding both left and right
+    if ( ply:KeyDown( IN_MOVELEFT ) and ply:KeyDown( IN_MOVERIGHT ) ) then return end
+    if ply:InVehicle() then return end
+    if ply:GetMoveType() == MOVETYPE_NOCLIP then return end
+
+    -- This determines which direction the player is trying to jump and sets dir accordingly
+    if ply:KeyDown( IN_MOVELEFT ) then
+        dir = 1
+    elseif ply:KeyDown( IN_MOVERIGHT ) then
+        dir = -1
+    end
+
+    -- Due to overwhelming popular demand
+    if wallkickInverted then
+        dir = -dir
+    end
+
+    -- If we're trying to go left or right
+    if dir ~= 0 then
+
+        -- Perform several traces to make it less likely that a hole in geometry would prevent a kick off
+        local traceCount = 5
+        local accumulatedNormal = Vector( 0, 0, 0 )
```

Comment:
`vector_origin` is easy for new developers to accidentally modify. We typically avoid using them just as a soft protection.

That being said, this could just be brought into a local at the top of the file